Inc/ST-LIB_HIGH/Protections/Boundary.hpp

```diff
@@ -16,246 +22,224 @@ enum ProtectionType : uint64_t {
 struct BoundaryInterface{
 public:
     virtual bool check_bounds() = 0;
-	virtual char* serialize(char* dst) = 0;
-	virtual int get_string_size() = 0;
+	HeapOrder* message;
 
+	void update_name(char* n){
+		name = n;
+		string_len = name.size();
+	}
+	virtual void update_error_handler_message( [[maybe_unused]] const char* err_message){
+	}
+	ProtectionType Protector = ERROR_HANDLER;
```

Comment:
??

Reply:
Boundary<ErrorHandler> needs this function, one approach is to implement it only in the Error Handler and then reinterpret_cast the BoundaryInterface* to Boundary<ErroHandler>* and call the method, I think this has some issues when compiling with -fno-rtti,
the other one is to just implement it in the base and override it in the Error Handler one